### PR TITLE
Implement copy artifact API

### PR DESCRIPTION
**Pull request by ywk253100**
Copy artifact into the repository from the specified artifact

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/api/artifact/controller.go [hunk 1]
```diff
@@ -64,6 +65,8 @@ type Controller interface {
 	GetByReference(ctx context.Context, repository, reference string, option *Option) (artifact *Artifact, err error)
 	// Delete the artifact specified by ID. All tags attached to the artifact are deleted as well
 	Delete(ctx context.Context, id int64) (err error)
+	// Copy the artifact whose ID is specified by "from" into the repository
+	Copy(ctx context.Context, repositoryID, from int64) (id int64, err error)
```
Comment (steven-zou) [inline src/api/artifact/controller.go]: from(int64) is better than from(*Artifact)?

`from(int64)` will also need to call `get()` internally

File: src/server/v2.0/handler/artifact.go [hunk 2]
```diff
@@ -36,15 +38,61 @@ func newArtifactAPI() *artifactAPI {
...
-	repoMgr repository.Manager
+	repoCtl repository.Controller
+}
+
+// TODO immutable, quota, readonly middlewares should cover this API
+func (a *artifactAPI) CopyArtifact(ctx context.Context, params operation.CopyArtifactParams) middleware.Responder {
+	if err := a.RequireProjectAccess(ctx, params.ProjectName, rbac.ActionCreate, rbac.ResourceArtifact); err != nil {
+		return a.SendError(ctx, err)
+	}
+	srcRepo, srcRef, err := parse(params.From)
```
Comment (reasonerjt) [inline src/server/v2.0/handler/artifact.go]: The API allows copy from artifact in other project, is there any security check?

File: src/api/artifact/controller.go [hunk 3]
```diff
@@ -65,6 +66,8 @@ type Controller interface {
...
```
Comment (reasonerjt) [inline src/api/artifact/controller.go]: why put src behind dst?

**Comment by reasonerjt** [comment]
The `Copy` takes two int64 parm the first one is artifact ID and the second one is repo ID， which can be confusing, but this is not blocker, I'll approve.

**Pull request merged**